client/blocks/nps/edit.js

```diff
@@ -57,11 +58,12 @@ const EditNpsBlock = ( props ) => {
 		try {
 			const { surveyId } = await updateNps(
 				tap(
-					pick( attributes, [
+					pick( { ...attributes, sourceLink }, [
```

Comment:
I'd say, either add it to the `data` object in the function below - similar to how the title fallback is set.
Or just rewrite the whole thing as:

```
const { feedbackQuestion, ratingQuestion, surveyId, title } = attributes;

updateNps( {
    feedbackQuestion,
    ratingQuestion,
    sourceLink,
    surveyId,
    title: title || ratingQuestion,
} );
```

The latter is probably the best for easy readability. I admit the `tap/pick` combo might've been a bit overkill.